multiversx/mx-chain-go
# New endpoint /node/waiting-epochs-left/:key

## Reasoning behind the pull request
- endpoint needed to provide the number of epochs left for a node in waiting state
  
## Proposed changes
- added new endpoint /waiting-epochs-left/:key as part of node group

## Testing procedure
- standard system test + check the new route(`/node/waiting-epochs-left/:key`) for the waiting nodes, then check that the information is updated at epoch change (on internal testnets, the returned value should be 1)
- check for an eligible node that this endpoint returns error

## Pre-requisites

Based on the [Contributing Guidelines](https://github.com/multiversx/mx-chain-go/blob/master/.github/CONTRIBUTING.md#branches-management) the PR author and the reviewers must check the following requirements are met:
- was the PR targeted to the correct branch?
- if this is a larger feature that probably needs more than one PR, is there a `feat` branch created?
- if this is a `feat` branch merging, do all satellite projects have a proper tag inside `go.mod`?
